src/Middleware/RedirectToLogin.php

```diff
@@ -29,19 +31,39 @@
  */
 class RedirectToLogin implements MiddlewareInterface
 {
+    private State $conf;
     private Horde_Registry $registry;
     private ResponseFactoryInterface $responseFactory;
-    public function __construct(Horde_Registry $registry, ResponseFactoryInterface $responseFactory)
+    public function __construct(Horde_Registry $registry, ResponseFactoryInterface $responseFactory, State $conf)
     {
         $this->registry = $registry;
         $this->responseFactory = $responseFactory;
+        $this->conf = $conf;
     }
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
         if ($request->getAttribute('HORDE_AUTHENTICATED_USER')) {
             return $handler->handle($request);
         }
-        $redirect = (string)Horde::Url($this->registry->getInitialPage('horde'), true);
-        return $this->responseFactory->createResponse(302)->withHeader('Location', $redirect);
+
+        $requestUrl = (string) $request->getUri();
+        $signedRequestUrl = Horde::signUrl($requestUrl);
+
+        // set baseurl: check if alternative login is set and use it as baseurl
+        $configArray = $this->conf->toArray();
+        $configArray['auth']['alternate_login'] ?? null;
+
+        if (!empty($alternateLogin)) {
+            $baseUrl = $alternateLogin;
+        }
+
+        // set baseurl: if no alternative login, use Horde login as baseurl
```

Comment:
Can you put that comment in the `else` block and keep the `if else` block together?

Reply:
ok done